core/domain/feedback_jobs_one_off.py

```diff
@@ -100,3 +102,88 @@ def _cache_last_nonempty_message_author_id(
             thread_model.last_nonempty_message_author_id = message_author_id
             return True
         return False
+
+
+class CleanUpFeedbackAnalyticsModelModelOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """One-off job to remove feedback analytics models for
+    deleted explorations.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [feedback_models.FeedbackAnalyticsModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+        exp_model = exp_models.ExplorationModel.get_by_id(
+            item.id)
+        if exp_model is None or exp_model.deleted:
+            yield ('Deleted Feedback Analytics Model', item.id)
+            item.delete()
+
+    @staticmethod
+    def reduce(key, values):
+        yield (key, values)
+
+
+class CleanUpGeneralFeedbackThreadModelOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """One-off job to clean up GeneralFeedbackThreadModel by:
+    1. Removing the model if the target model for which feedback was created is
+    deleted. Target model can be exploration, question, skill or topic.
+    2. Update the last updated date time if it less than the creation time.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [feedback_models.GeneralFeedbackThreadModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+        target_model = (
+            prod_validators.TARGET_TYPE_TO_TARGET_MODEL[
+                item.entity_type].get_by_id(item.entity_id))
+        if target_model is None or target_model.deleted:
+            yield ('Deleted GeneralFeedbackThreadModel', item.id)
+            item.delete()
+            return
+
+        if item.created_on > item.last_updated:
+            item.put()
+            yield (
+                'Updated last_updated field for '
+                'GeneralFeedbackThreadModel', item.id)
+
+    @staticmethod
+    def reduce(key, values):
+        yield (key, values)
+
+
+class CleanUpGeneralFeedbackMessageModelOneOffJob(
```

Comment:
Adding a note to confirm - that this issue does not exist anymore, so we don't need any additional fix.